### PR TITLE
164 new right pane UI

### LINKED ISSUE #164: Update right pane UI according to Figma design

### PR DESCRIPTION
**Pull request by GavinSin**
- [X]  This pull request is complete
- [ ]  This pull request is a work in progress

### This PR aims to implement:

- An overview of total balances component in Right Panel
- A clickable filter icon that filters transaction list
- New layout for transaction card component according to design doc

---

### Acceptance Criteria

- When transaction list is empty, the total balance shows 0. 
- Positive amounts are display in green text, and prepended with "owes you".
- Negative amounts are display in red text, and prepended with "you owe".
- Date is display in two rows, first row is month in 3 chars i.e. "Jan", second row is day i.e. "1", "24".

### Design Doc
https://www.figma.com/design/5AwS9EGiBLAzjAhTAxQ5Vb/spleetwise?node-id=0-1&t=k54E8C1Swad4P4o9-1

---

closes #164

### CODE REVIEW

**Comment by GavinSin** [comment]
Related issues to continue on:
- By default, transaction is unmark. If transaction is marked, a done icon appears right of person's name.
  - Dependency: `markTxn`, `unmarkTxn` are created, and there is marked field in `Transaction` object.
  - Issues: #159, #180 
- Custom filter menu where user can filter by inputted range, by person name (multiple options checkbox), by amount (by inputted range)
  - Issue: #181

**Review by Dino-Nuggies — CHANGES_REQUESTED** [review]
There are some incorrect JavaDocs that needs to be addressed.

Additionally, the total balance window doesnt seem to auto update every time a new transaction is added

File: src/main/java/spleetwaise/transaction/ui/RightPanel.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    private StackPane transactionListPanelPlaceholder;
+    @FXML
+    private Label youOwnLabel;
+    @FXML
+    private Label ownYouLabel;
+    @FXML
+    private Label filterIcon;
+
+    /**
+     * Creates a {@code PersonCode} with the given {@code Person} and index to display.
```
[the remark this answers is not in the document]
Comment (GavinSin) [reply src/main/java/spleetwaise/transaction/ui/RightPanel.java]: I have corrected!

**Comment by GavinSin** [comment]
> There are some incorrect JavaDocs that needs to be addressed.
> 
> Additionally, the total balance window doesnt seem to auto update every time a new transaction is added

Thanks for the catch, I have fixed the issue.

File: src/main/java/spleetwaise/transaction/ui/TransactionCard.java [hunk 2]
```diff
@@ -22,29 +23,38 @@ public class TransactionCard extends UiPart<Region> {
...
     @FXML
     private Label description;
     @FXML
-    private Label date;
+    private Label month;
+    @FXML
+    private Label day;
 
     /**
      * Creates a {@code PersonCode} with the given {@code Person} and index to display.
```
[the remark this answers is not in the document]
Comment (GavinSin) [reply src/main/java/spleetwaise/transaction/ui/TransactionCard.java]: Resolved!

**Review by Dino-Nuggies — APPROVED** [review]
LGMT, problem with right pane not updating being fixed too

**Pull request merged**